libs/features/personalization/personalization.js

```diff
@@ -271,30 +272,98 @@ function parsePlaceholders(placeholders, config, selectedVariantName = '') {
   return config;
 }
 
-function getPersonalizationVariant(manifestPath, variantNames = [], variantLabel = null) {
+const fetchEntitlements = async () => {
+  const [{ default: getUserEntitlements }] = await Promise.all([
+    import('../../blocks/global-navigation/utilities/getUserEntitlements.js'),
+    loadIms(),
+  ]);
+  return getUserEntitlements();
+};
+
+export const getEntitlements = (() => {
+  let ents;
+  return (async () => {
```

Comment:
if `ents` already defined, we can return it without returning an async function.

Reply:
The `return` async function here is assigned to `getEntitlements` (it's an IIFE)